Helper/ApiClient.php

```diff
@@ -29,56 +29,99 @@ public function __construct(\Magento\Store\Model\StoreManagerInterface $storeMan
     $this->_imgHelper = $imgHelper;
   }
 
- public function prepareProductsData($order) 
+  public function prepareProductsData($order) 
   {
-        $this->_storeManager->setCurrentStore($order->getStoreId());
-        $products = $order->getAllItems(); //filter out simple products
-        $products_arr = array();
+    $this->_storeManager->setCurrentStore($order->getStoreId());
+    $objectManager = \Magento\Framework\App\ObjectManager::getInstance();
+    $groupedProductModel = $objectManager->create('\Magento\GroupedProduct\Model\Product\Type\Grouped');
+    $productModel = $objectManager->create('\Magento\Catalog\Model\Product');
+    $productCollection = $objectManager->create('\Magento\Catalog\Model\ResourceModel\Product\CollectionFactory');
+    $store = $objectManager->get('Magento\Store\Model\StoreManagerInterface')->getStore();
+    
+    $products = $order->getAllVisibleItems();
+    $products_arr = array();
+    $product_data = array();
+	$specs_data = array();
+        $j = 0;
         foreach ($products as $item) {
-			if($item->getData('product_type')!='configurable'){
-				$full_product = $this->_productRepository->get($item->getSku());
-				$parentId = $item->getProduct()->getId();
-				if (!empty($parentId)) {
-					$full_product = $this->_productRepository->load($parentId);
-				}
-				$specs_data = array();
-				$product_data = array();
-				$product_data['name'] = $full_product->getName();
-				$product_data['url'] = '';
-				$product_data['image'] = '';
-				try {
-					$product_data['url'] = $full_product->getUrlInStore(array('_store' => $order->getStoreId()));
-					$product_data['image'] = $this->_imgHelper->init($full_product, 'product_base_image')->getUrl();
-					if ($full_product->getUpc()) {
-						$specs_data['upc'] = $full_product->getUpc();
-					}
-					if ($full_product->getIsbn()) {
-						$specs_data['isbn'] = $full_product->getIsbn();
-					}
-					if ($full_product->getBrand()) {
-						$specs_data['brand'] = $full_product->getBrand();
-					}
-					if ($full_product->getMpn()) {
-						$specs_data['mpn'] = $full_product->getMpn();
-					}
-					if ($full_product->getSku()) {
-						$specs_data['external_sku'] = $full_product->getSku();
-					}
-					if (!empty($specs_data)) {
-						$product_data['specs'] = $specs_data;
-					}
-				} catch (\Exception $e) {
-					$this->_logger->addDebug('ApiClient prepareProductsData Exception' . json_encode($e));
-				}
-				$rawdescription =  str_replace(array('\'', '"'), '', $full_product->getDescription()); 
-				$description =  $this->_escaper->escapeHtml(strip_tags($rawdescription));
-				$product_data['description'] = $description;
-				$product_data['price'] = $item->getPrice();
-				$products_arr[$full_product->getId()] = $product_data;
-			}
+            try {
+                $productID = $item->getProduct()->getId();
+                if ($item->getData('product_type') == 'simple') {
```

Comment:
1. get product type once in variable at the begining , product_type = $item->getData('product_type')